### ISSUE TITLE
is there a way to see test log?

### ISSUE THREAD

**Issue by jarivk** [bats-core/bats-core#267]
I am not able to see the logs of my test command specified in "run". How to see that? I would like it to be printed on terminal.

**Comment by dimo414** [comment]
`echo "$output"`?

**Comment by jarivk** [comment]
It doesn't work. Text is printed only if there is a test failure.
Anyway I think it is fine. So can close the the issue.

**Comment by dimo414** [comment]
It's by-design that test output isn't printed on success (it would be much noisier otherwise), but test artifacts could still be preserved elsewhere. I don't *think* there's any support for that currently, but it would be a reasonable FR to try to flesh out preserving artifacts more consistently.

**Comment by jarivk** [comment]
Yes, That will be good. Reopening the bug. Feel free to close it.

**Comment by xmik** [comment]
Adding:
```
echo "# output: $output" >&3
```
will print the output on both: failure and success. It is written in the readme: https://github.com/bats-core/bats-core#printing-to-the-terminal . However, it gets printed in red (at least for me).

**Comment by martin-schulze-vireso** [comment]
Since this has cropped up from time to time I'd like to adress it. But first we'd need to flesh out what should be done.
I think it would amount to some kind of `--verbose` mode. Internally that could be reflected by a `BATS_VERBOSE` variable that can be turned on/off per test.

**Comment by dimo414** [comment]
How about instead adding an `--artifacts_dir` flag that writes all output to the given directory. That way the terminal output isn't flooded and the output can be separated per-test?

**Comment by martin-schulze-vireso** [comment]
I find the term artifact to be not quite fitting. Apart from the separation into different files there is not much difference to --report-formatter.
